src/manage_sql_nvts.c

```diff
@@ -1437,19 +1436,21 @@ update_nvts_from_vts (entity_t *get_vts_response,
 
       insert_nvt (nvti);
 
+      preferences = NULL;
       if (update_preferences_from_vt (vt, nvti_oid (nvti), &preferences))
         {
           sql_rollback ();
           return;
         }
+      sql ("DELETE FROM nvt_preferences WHERE name LIKE '%s:%%';",
```

Comment:
Would it help with the performance of the first update to check outside the loop if there are any preferences and skip this `DELETE` if that's the case? Otherwise it would be run without any effect for every NVT.

Reply:
Usually this only considers NVTs added/updated since the last sync, so I don't think the DELETEs will be a problem.  Also not sure if there's an easy SQL statement to check this.

But something to keep in mind if we see performance problems.